### PR TITLE
Fix cluster pause webhook to set the temporary pause annotation on cluster's TKR label change

### LINKED ISSUE #2552: Addons-controller stalls cluster upgrade
**Bug description**
Cluster pause webhook should compare cluster.metadata.labels[run.tanzu.vmware.com/tkr] and set paused if there are changes

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [X] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by maralavi**
### What this PR does / why we need it
Fixes cluster pause webhook to set the temporary pause annotation on cluster's TKR label change

### Which issue(s) this PR fixes

Fixes #2552 

### Describe testing done for PR
Added unit tests, also tested manually in the cluster with cluster Create

### Release note

```release-note
package-based-lcm: Fix cluster pause webhook to set the temporary pause annotation on cluster's TKR label change
```

### PR Checklist

- [X] Squash the commits into one or a small number of logical commits
      
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: packages/addons-manager/bundle/config/upstream/webhook-manifests.lib.yaml [hunk 1]
```diff
@@ -87,6 +87,7 @@ webhooks:
         path: /mutate-run-tanzu-vmware-com-v1alpha3-clusterbootstrap
     failurePolicy: Fail
     name: clusterbootstrap.mutating.vmware.com
+    reinvocationPolicy: IfNeeded
     rules:
```
Comment (vijaykatam) [inline packages/addons-manager/bundle/config/upstream/webhook-manifests.lib.yaml:91]: @shyaamsn / @yharish991 - Are there other places to update for consumption by TKGS?

File: pkg/v1/webhooks/cluster_pause_webhook.go [hunk 2]
```diff
@@ -52,17 +65,19 @@ func (wh *ClusterPause) Default(ctx context.Context, obj runtime.Object) error {
...
-			}
-			// Use the desired TKR version as label value, ClusterBootstrap will unset
-			cluster.Annotations[constants.ClusterPauseLabel] = cluster.Spec.Topology.Version
+	if currentCluster.Labels != nil {
+		currentTkrVersion, ok = currentCluster.Labels[v1alpha3.LabelTKR]
+	}
+
+	// Add pause to cluster if the cluster.Labels["run.tanzu.vmware.com/tkr"] changes
+	// The cluster pause state will be unset by ClusterBootstrap controller after it rolls out package updates
+	if currentCluster.Labels == nil || !ok || currentTkrVersion != tkrVersion {
```
Comment (vijaykatam) [inline pkg/v1/webhooks/cluster_pause_webhook.go:74]: Could we remove `ok` check as we are already comparing `currentTkrVersion != tkrVersion`? Or are you trying to guard against someone removing tkr label? If latter please rename `ok` to `tkrLabelFound`

Comment (vijaykatam) [inline pkg/v1/webhooks/cluster_pause_webhook.go:74]: Could we remove `ok` check as we are already comparing `currentTkrVersion != tkrVersion`?

Comment (maralavi) [reply pkg/v1/webhooks/cluster_pause_webhook.go]: Yes, the latter case. Changed to use ``tkrLabelFound``` instead

**Review by vijaykatam — APPROVED** [review]
LGTM. Lets make codecov happy and should be good to merge.

File: packages/addons-manager/bundle/config/upstream/webhook-manifests.lib.yaml (hunk 1 above)
Comment (shyaamsn) [reply packages/addons-manager/bundle/config/upstream/webhook-manifests.lib.yaml:91]: If its not parametrized using values.yaml then no need to update it in other packages.

File: pkg/v1/webhooks/cluster_pause_webhook.go [hunk 3]
```diff
@@ -36,33 +37,46 @@ var _ webhook.CustomDefaulter = &ClusterPause{}
 
 // Default satisfies the defaulting webhook interface.
 func (wh *ClusterPause) Default(ctx context.Context, obj runtime.Object) error {
+	var cluster *clusterv1.Cluster
+	var tkrVersion, currentTkrVersion string
+	var tkrLabelFound bool
+
```
Comment (maralavi) [inline pkg/v1/webhooks/cluster_pause_webhook.go:43]: But these are variable definitions and can not be covered by a test!

**Comment by maralavi** [comment]
@vijaykatam I believe it is complaining because the tests are added in another directory and it can not find those!

File: pkg/v1/webhooks/cluster_pause_webhook.go (hunk 3 above)
Comment (maralavi) [reply pkg/v1/webhooks/cluster_pause_webhook.go:43]: All other lines are also already covered in addons/controllers/cluster_pause_webhook_test.go

**Pull request merged**